update/factory/exportHandlerFactory.go

```diff
@@ -0,0 +1,259 @@
+package factory
+
+import (
+	"path"
+
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/data/typeConverters"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/epochStart"
+	"github.com/ElrondNetwork/elrond-go/epochStart/notifier"
+	"github.com/ElrondNetwork/elrond-go/epochStart/shardchain"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	storageFactory "github.com/ElrondNetwork/elrond-go/storage/factory"
+	"github.com/ElrondNetwork/elrond-go/storage/storageUnit"
+	"github.com/ElrondNetwork/elrond-go/update"
+	"github.com/ElrondNetwork/elrond-go/update/files"
+	"github.com/ElrondNetwork/elrond-go/update/genesis"
+	"github.com/ElrondNetwork/elrond-go/update/sync"
+)
+
+type ArgsExporter struct {
+	Marshalizer              marshal.Marshalizer
+	Hasher                   hashing.Hasher
+	HeaderValidator          epochStart.HeaderValidator
+	Uint64Converter          typeConverters.Uint64ByteSliceConverter
+	DataPool                 dataRetriever.PoolsHolder
+	StorageService           dataRetriever.StorageService
+	RequestHandler           update.RequestHandler
+	ShardCoordinator         sharding.Coordinator
+	Messenger                p2p.Messenger
+	ActiveTries              state.TriesHolder
+	ExportFolder             string
+	ExportTriesStorageConfig config.StorageConfig
+	ExportTriesCacheConfig   config.CacheConfig
+	ExportStateStorageConfig config.StorageConfig
+}
+
+type exportHandlerFactory struct {
+	marshalizer              marshal.Marshalizer
+	hasher                   hashing.Hasher
+	headerValidator          epochStart.HeaderValidator
+	uint64Converter          typeConverters.Uint64ByteSliceConverter
+	dataPool                 dataRetriever.PoolsHolder
+	storageService           dataRetriever.StorageService
+	requestHandler           update.RequestHandler
+	shardCoordinator         sharding.Coordinator
+	messenger                p2p.Messenger
+	activeTries              state.TriesHolder
+	exportFolder             string
+	exportTriesStorageConfig config.StorageConfig
+	exportTriesCacheConfig   config.CacheConfig
+	exportStateStorageConfig config.StorageConfig
+}
+
+func NewExportHandlerFactory(args ArgsExporter) (*exportHandlerFactory, error) {
+	if check.IfNil(args.ShardCoordinator) {
+		return nil, sharding.ErrNilShardCoordinator
+	}
+	if check.IfNil(args.Hasher) {
+		return nil, sharding.ErrNilHasher
+	}
+	if check.IfNil(args.Marshalizer) {
+		return nil, data.ErrNilMarshalizer
+	}
```

Comment:
Aren't checks for `IfNil` for the remaining fields of `args` necessary?

Reply:
done.